ros2doctor/ros2doctor/verb/call.py

```diff
@@ -0,0 +1,260 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from argparse import ArgumentTypeError
+import socket
+import struct
+import threading
+import time
+
+import rclpy
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.node import Node
+from ros2doctor.verb import VerbExtension
+
+from std_msgs.msg import String
+
+DEFAULT_GROUP = '225.0.0.1'
+DEFAULT_PORT = 49150
+
+
+def positive_int(string: str) -> int:
+    try:
+        value = int(string)
+    except ValueError:
+        value = -1
+    if value <= 0:
+        raise ArgumentTypeError('value must be a positive integer')
+    return value
+
+
+class CallVerb(VerbExtension):
+    """
+    Check network connectivity between multiple hosts.
+
+    This command can be invoked on multiple hosts to confirm that they can talk to each other
+    by using talker/listener, multicast send/receive to check topic discovering and
+    UDP communication.
+    This command outputs a summary table of msgs statistics at a custom rate(Hz).
+    """
+
+    def add_arguments(self, parser, cli_name):
+        parser.add_argument(
+            'topic_name', nargs='?', default='/canyouhearme',
+            help="Name of ROS topic to publish to (default: '/canyouhearme')")
+        parser.add_argument(
+            'time_period', nargs='?', default=0.1,
+            help='Time period to publish/send one message (default: 0.1s)')
+        parser.add_argument(
+            'duration', nargs='?', default=20, type=positive_int,
+            help='How long this process runs (default: 20s)')
+        parser.add_argument(
+            '-r', '--rate', metavar='N', type=float, default=1.0,
+            help='Rate in Hz to print summary table (default: 1.0)')
+        parser.add_argument(
+            '--ttl', type=positive_int,
+            help='TTL for multicast send (default: None)')
+
+    def main(self, *, args):
+        global summary_table
+        summary_table = SummaryTable()
+        rclpy.init()
+        executor = SingleThreadedExecutor()
+        pub_node = Talker(args.topic_name, args.time_period)
+        sub_node = Listener(args.topic_name)
+        executor.add_node(pub_node)
+        executor.add_node(sub_node)
+        try:
+            prev_time = time.time()
+            timeout = time.time() + args.duration
+            # pub/sub thread
+            exec_thread = threading.Thread(target=executor.spin)
+            exec_thread.start()
+            while time.time() < timeout:
```

Comment:
Sorry if I gave you the wrong impression before, but I think looping indefinitely until a keyboard interrupt is fine here. Unless you think there's a use-case for only running for a limited duration, in which case we should still have the option to run indefinitely. 
```suggestion
            while True:
```